app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt

```diff
@@ -0,0 +1,306 @@
+          var progress = 0f
+          val startTime = System.currentTimeMillis()
+
+          while (progress < 1f) {
+            progress = (System.currentTimeMillis() - startTime).toFloat() / spinDuration
+            val easeOut = 1f - (1f - progress) * (1f - progress) * (1f - progress) * (1f - progress)
+            rotation = currentRotation + (targetRotation - currentRotation) * easeOut
+            delay(16)
+          }
+
+          rotation = targetRotation % 360
+          isSpinning = false
+          pauseAfterSpin = true
+          delay(2000)
+          pauseAfterSpin = false
+
+          onSegmentLanded?.invoke(segments[getSegmentAtPointer(rotation)].name)
+          Log.d(
+              "FAVORITE",
+              "Spin finished on segment ${getSegmentAtPointer(rotation)} (${segments[getSegmentAtPointer(rotation)].name})")
+        }
+      }
+    }
+  }
+
+  Box(modifier = modifier.testTag("wheel_view_container")) {
+    Canvas(modifier = Modifier.testTag("wheel_canvas").aspectRatio(1f)) {
+      val centerX = size.width / 2f
+      val centerY = size.height / 2f
+      val radius = minOf(size.width, size.height) / 2f * 0.9f
+
+      rotate(rotation, Offset(centerX, centerY)) {
+        val segmentAngle = 360f / segments.size
+        segments.forEachIndexed { index, segment ->
+          drawArc(
+              color = segment.color,
+              startAngle = index * segmentAngle,
+              sweepAngle = segmentAngle,
+              useCenter = true,
+              topLeft = Offset(centerX - radius, centerY - radius),
+              size = Size(radius * 2, radius * 2))
+
+          val startAngle = index * segmentAngle
+          val middleAngle = Math.toRadians((startAngle + segmentAngle / 2.0))
+          val textRadius = radius * 0.6f
+          val textX = centerX + cos(middleAngle).toFloat() * textRadius
+          val textY = centerY + sin(middleAngle).toFloat() * textRadius
+
+          rotate(startAngle + segmentAngle / 2 + 90, Offset(textX, textY)) {
+            drawText(
+                textMeasurer = textMeasurer,
+                text = segment.name,
+                topLeft = Offset(textX - 25f, textY - 10f),
+                style =
+                    TextStyle(
+                        color = Color.White,
+                        fontSize = (radius * 0.06f).sp,
+                        textAlign = TextAlign.Center))
+          }
+        }
+      }
+
+      drawPath(
+          path =
+              Path().apply {
+                val pointerWidth = radius * 0.1f
+                val pointerHeight = radius * 0.1f
+                moveTo(centerX, centerY - radius + pointerHeight * 1.5f)
+                lineTo(centerX - pointerWidth / 2, centerY - radius - pointerHeight / 2)
+                lineTo(centerX + pointerWidth / 2, centerY - radius - pointerHeight / 2)
+                close()
+              },
+          color = Color.Red)
+    }
+
+    segments.forEachIndexed { index, segment ->
```

Comment:
I'm not sure what these empty boxes are used for?

Reply:
To add testTags where testTags cant be added.

Reply:
I tried commenting them and removing the tests where they were used and coverage stayed very decent. Since they are just invisible boxes, these tests were just covering the "invisible boxes". I don't think they are needed:
![image](https://github.com/user-attachments/assets/7dd64959-6201-4490-934b-82a55c162119)